tests/filecheck/lit.cfg

```diff
@@ -5,6 +5,7 @@ config.name = "xDSL"
 config.test_format = lit.formats.ShTest()
 config.suffixes = ['.test', '.xdsl', '.mlir']
 
+config.test_exec_root = os.getcwd()
```

Comment:
Maybe I am missing something. From the [docu](https://llvm.org/docs/CommandGuide/lit.html#test-suites):
```
test_exec_root: For out-of-dir builds, the path to the test suite root inside the object directory. This is where tests will be run and temporary output files placed.
```

This is all nice and good for the tests defined in `tests/filecheck/`. The coverage files for those are generated in the expected `xdsl` folder. However, for every subfolder, say `tests/filecheck/irdl-printer/`, the coverage files are generate in a *new* folder `xdsl/irdl-printer`. Why does this happen? If I print the current working directory from within `xdsl-opt` I can clearly see that it is not executed in the `test_exec_root` but in a new directory `test_exec_root/*whatever is after tests/filecheck*`.

I will give up for today. I'll try to look into it again tomorrow and start to find workaround otherwise...

Reply:
So you mean that there is a `filecheck` that is appended at the end? That is weird :/ Maybe a bug in lit?

Reply:
Not quite, there are subdirectories in the `filecheck` folder created and appended.

Anywho, I will try to workaround this. It will probably result in the pytest config and the lit config not being at the same place, which is unfortunate, but probably okay for a first version. I will add a TODO though.

Reply:
I have now split the configs again. Therefore, lit does not have to execute in the proper folder and I can collect the coverage data in the combine call.

I'll look into defining aliases in lit. I think it would be quite lit (badumtss) if we could give it a flag that generates coverage. Otherwise, running tests locally will create a ton of unnecessary files.